owncloudData/src/test/java/com/owncloud/android/data/files/repository/OCFileRepositoryTest.kt

```diff
@@ -25,382 +26,2010 @@ import com.owncloud.android.data.files.datasources.LocalFileDataSource
+                accountName = fileWithConflict.owner
+            )
+            remoteFileDataSource.deleteFile(
+                remotePath = fileWithConflict.remotePath,
+                accountName = fileWithConflict.owner,
+                spaceWebDavUrl = OC_SPACE_PERSONAL.root.webDavUrl
+            )
+            localFileDataSource.cleanConflict(fileWithConflict.id!!)
+            localStorageProvider.deleteLocalFile(fileWithConflict)
+            localFileDataSource.deleteFile(fileWithConflict.id!!)
+        }
+    }
+
+    @Test
+    fun `deleteFiles removes a file and its conflict but only from local correctly`() {
+        every {
+            localStorageProvider.deleteLocalFile(fileWithConflict)
+        } returns true
+
+        ocFileRepository.deleteFiles(listOf(fileWithConflict), true)
+
+        verify(exactly = 1) {
+            localSpacesDataSource.getWebDavUrlForSpace(
+                spaceId = fileWithConflict.spaceId,
+                accountName = fileWithConflict.owner
+            )
+            localFileDataSource.cleanConflict(fileWithConflict.id!!)
+            localStorageProvider.deleteLocalFile(fileWithConflict)
+            localFileDataSource.saveFile(fileWithConflict.copy(storagePath = null, etagInConflict = null, lastUsage = null, etag = null))
+        }
+    }
+
+    @Test
+    fun `deleteFiles removes a folder recursively from local and remote correctly`() {
+        every {
+            localFileDataSource.getFolderContent(OC_PARENT_FOLDER_WITH_SPACE_ID.id!!)
+        } returns listOf(OC_FILE_WITH_SPACE_ID)
+        every {
+            localStorageProvider.deleteLocalFile(OC_FILE_WITH_SPACE_ID)
+        } returns true
+
+        ocFileRepository.deleteFiles(listOf(OC_PARENT_FOLDER_WITH_SPACE_ID), false)
+
+        verify(exactly = 1) {
+            localSpacesDataSource.getWebDavUrlForSpace(
+                spaceId = OC_PARENT_FOLDER_WITH_SPACE_ID.spaceId,
+                accountName = OC_PARENT_FOLDER_WITH_SPACE_ID.owner
+            )
+            remoteFileDataSource.deleteFile(
+                remotePath = OC_PARENT_FOLDER_WITH_SPACE_ID.remotePath,
+                accountName = OC_PARENT_FOLDER_WITH_SPACE_ID.owner,
+                spaceWebDavUrl = OC_SPACE_PERSONAL.root.webDavUrl
+            )
+            localFileDataSource.getFolderContent(OC_PARENT_FOLDER_WITH_SPACE_ID.id!!)
+            localStorageProvider.deleteLocalFile(OC_FILE_WITH_SPACE_ID)
+            localFileDataSource.deleteFile(OC_FILE_WITH_SPACE_ID.id!!)
+            localStorageProvider.deleteLocalFolderIfItHasNoFilesInside(OC_PARENT_FOLDER_WITH_SPACE_ID)
+            localFileDataSource.deleteFile(OC_PARENT_FOLDER_WITH_SPACE_ID.id!!)
+        }
+    }
+
+    @Test
+    fun `renameFile renames a file correctly`() {
+        val newName = "image3.jpt"
+        val newRemotePath = "/Folder/image3.jpt"
+        val newStoragePath = "/local/storage/path/username@demo.owncloud.com/Folder/image3.jpt"
```

Comment:
A global constant better?

Reply:
Didn't make it a global variable because we have 3 different cases (although they are very similar):

Case 1: `/local/storage/path/username@demo.owncloud.com/Folder/Photos/image2.jpt`
Case 2: `/local/storage/path/username@demo.owncloud.com/Folder/Photos/image2 (1).jpt`
Case 3: `/local/storage/path/username@demo.owncloud.com/Folder/image3.jpt`

Each of them used for a different purpose 😀

Reply:
You are right, they are different. I thought they were the same :)